File: backend/src/main/java/com/qme/qmeapp/queue/dto/QueueDetailsResponse.java
package com.qme.qmeapp.queue.dto;

import com.qme.qmeapp.organization.dto.OrganizationResponse;
import lombok.Builder;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
public class QueueDetailsResponse {
    private UUID entryId;
    private OrganizationResponse organization;
    private Integer queueNumber;
    private Integer positionInLine;
    private String computedStatus; // "BEING_SERVED", "NEXT_IN_LINE", "WAITING"
    private Integer currentlyServingNumber;
    private Integer estimatedWaitTime;
    private OffsetDateTime joinedAt;
}

File: mobile/app/src/main/java/com/qme/mobile/data/model/response/QueueDetailsResponse.kt
package com.qme.mobile.data.model.response

data class QueueDetailsResponse(
    val entryId: String,
    val organization: OrganizationResponse,
    val queueNumber: Int,
    val positionInLine: Int,
    val computedStatus: String,   // BEING_SERVED | NEXT_IN_LINE | WAITING
    val currentlyServingNumber: Int?,
    val estimatedWaitTime: Int?,
    val joinedAt: String?
)
